quickwit/quickwit-janitor/src/garbage_collection.rs

```diff
@@ -101,26 +101,26 @@ pub async fn run_garbage_collect(
     let grace_period_timestamp =
         OffsetDateTime::now_utc().unix_timestamp() - staged_grace_period.as_secs() as i64;
 
-    let deletable_staged_splits: Vec<SplitMetadata> = protect_future(
-        ctx_opt,
-        metastore.list_splits(index_id, SplitState::Staged, None, None),
-    )
-    .await?
-    .into_iter()
-    // TODO: Update metastore API and push this filter down.
-    .filter(|meta| meta.update_timestamp < grace_period_timestamp)
-    .map(|meta| meta.split_metadata)
-    .collect();
+    let filter = ListSplitsQuery::for_index(index_id)
+        .with_split_state(SplitState::Staged)
+        .updated_after(grace_period_timestamp);
```

Comment:
The current filter is `meta.update_timestamp < grace_period_timestamp`. So, We want the split's last update timestamp to be anterior to the grace period timestamp, which I believe translates to `updated_before`. If this hasn't been caught by any unit test, please add one.

At the end of the day, I'm not a fan of the before/after API: I'd rather use the well-known operators `eq`, `gt`, `gte`, ...:
`.update_timestamp_lt(grace_period_timestamp)`;

Reply:
Implemented in the design